eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/sorted/immutable/AbstractImmutableSortedBag.java

```diff
@@ -322,6 +323,54 @@ public ImmutableShortList collectShort(ShortFunction<? super T> shortFunction)
     }
 
     @Override
+    public <S, Z> OrderedIterable<Z> zipWith(Iterable<S> that, Function2<T, S, Z> function)
+    {
+        MutableList<Z> list = FastList.newList();
+        Iterator<S> iterator = that.iterator();
+
+        this.forEachWithOccurrences((each, parameter) -> {
+            for (int i = 0; i < parameter; i++)
+            {
+                if (iterator.hasNext())
+                {
+                    list.add(function.apply(each, iterator.next()));
+                }
+            }
+        });
+        return list.toImmutable();
+    }
+
+    @Override
+    public <S, Z, R extends Collection<Z>> R zipWith(Iterable<S> that, Function2<T, S, Z> function, R target)
+    {
+        Iterator<S> iterator = that.iterator();
+
+        if (target instanceof MutableBag)
+        {
+            MutableBag<Z> targetBag = (MutableBag<Z>) target;
+            this.forEachWithOccurrences((each, occurrences) -> {
+                if (iterator.hasNext())
+                {
+                    targetBag.addOccurrences(function.apply(each, iterator.next()), occurrences);
```

Comment:
Is there a test that covers this line? If I'm reading it right, it looks like the target bag will wind up with size^2 number of elements. I realize you're replicating the pattern from zip and zipWith, so I want to make sure all three are covered by tests.